addon/services/scroller.js

```diff
@@ -32,14 +32,19 @@ export default Service.extend({
   },
 
   // scroll to an element by id
-  scrollToElementId: task(function * (elementId, options = {}) {
+  scrollToElementId(elementId, options = {}) {
     let element = document && document.getElementById(elementId);
     assert(`An element with the id: '${elementId}' could not be found in the DOM. Be sure to check that it has been rendered before attempting to scroll to it.`, element)
-    yield this.get('scrollToElement').perform(element, options);
+    return this.scrollToElement(element, options);
+  },
+
+  // task version so you can use the perform helper
+  scrollToElementIdTask: task(function * () {
+    yield this.scrollToElementId(...arguments);
```

Comment:
I think I'd implement this in reverse. Have most of the code logic in the task and have the method do nothing more than perform the task.

```js
  // scroll to an element by id
  scrollToElementId(elementId, options = {}) {
    this.get('scrollToElementIdTask').perform(elementId, options);
  },

  // task version so you can use the perform helper
  scrollToElementIdTask: task(function* (elementId, options = {}) {
    let element = document && document.getElementById(elementId);
    assert(`An element with the id: '${elementId}' could not be found in the DOM. Be sure to check that it has been rendered before attempting to scroll to it.`, element)
    yield this.scrollToElement(element, options);
  }),
```